StackoverflowChatbot/Discord.cs

```diff
@@ -88,6 +89,28 @@ private static Task ClientRecieved(SocketMessage arg)
 			return Task.CompletedTask;
 		}
 
+        private static string BuildSoMessage(SocketGuildUser user, Config.Base config, SocketMessage arg)
+        {
+			var displayname = string.IsNullOrEmpty(user.Nickname) ? user.Username : user.Nickname;
+            var messageStart = $@"\[**[{displayname}]({config.DiscordInviteLink})**]";
+			var messageContent = arg.Content;
+			foreach (var mentionedUser in arg.MentionedUsers)
+			{
+				messageContent.Replace(mentionedUser.Mention, $"@{mentionedUser.Username}");
+			}
+			foreach (var mentionedRoles in arg.MentionedRoles)
```

Comment:
What's the difference between roles and users in this context? I thought roles were like user groups.

Reply:
Roles are indeed groups, and users are users.

I distinguished between the two since groups are not available on SO (except you want to replace the group with an individual ping for every user in the group), and users might have the same name on SO.

Reply:
Good enough for me